extensions/objects/templates/image-editor.component.ts

```diff
@@ -0,0 +1,1155 @@
+
+    let newImageFile;
+
+    if (mimeType === 'data:image/gif') {
+      let successCb = obj => {
+        this.validateProcessedFilesize(obj.image);
+        newImageFile = (
+          this.imageUploadHelperService.convertImageDataToImageFile(
+            obj.image));
+        this.updateDimensions(newImageFile, obj.image, width, height);
+        document.body.style.cursor = 'default';
+      };
+      let processFrameCb = this.getCroppedGIFDataAsync.bind(
+        null, x1, y1, width, height);
+      this.processGIFImage(
+        imageDataURI, width, height, processFrameCb, successCb);
+    } else if (mimeType === 'data:image/svg+xml') {
+      // Check point 2 in the note before imports and after fileoverview.
+      const imageData = this.imgData || (
+        this.data.metadata.uploadedImageData as string);
+      newImageFile = (
+        this.imageUploadHelperService.convertImageDataToImageFile(
+          this.data.metadata.uploadedImageData as string));
+      this.updateDimensions(newImageFile, imageData, width, height);
+    } else {
+      // Generate new image data and file.
+      const newImageData = this.getCroppedImageData(
+        // Check point 2 in the note before imports and after fileoverview.
+        this.imgData || this.data.metadata.uploadedImageData,
+        x1, y1, width, height);
+      this.validateProcessedFilesize(newImageData);
+
+      newImageFile = (
+        this.imageUploadHelperService.convertImageDataToImageFile(
+          newImageData));
+      this.updateDimensions(newImageFile, newImageData, width, height);
+    }
+  }
+
+  updateDimensions(
+      newImageFile: File,
+      newImageData: string,
+      width: number,
+      height: number): void {
+    // Update image data.
+    this.data.metadata.uploadedFile = newImageFile;
+    this.data.metadata.uploadedImageData = newImageData;
+    this.imgData = newImageData;
+    this.data.metadata.originalWidth = width;
+    this.data.metadata.originalHeight = height;
+    // Check point 1 in the note before imports and after fileoverview.
+    this.data = {...this.data};
+    // Re-calculate the dimensions of the base image and reset the
+    // coordinates of the crop area to the boundaries of the image.
+    const dimensions = this.calculateTargetImageDimensions();
+    this.cropArea = {
+      x1: 0,
+      y1: 0,
+      x2: dimensions.width,
+      y2: dimensions.height
+    };
+  }
+
+  cancelCropImage(): void {
+    const dimensions = this.calculateTargetImageDimensions();
+    this.cropArea.x1 = 0;
+    this.cropArea.y1 = 0;
+    this.cropArea.x2 = dimensions.width;
+    this.cropArea.y2 = dimensions.height;
+  }
+
+  getImageSizeHelp(): string | null {
+    const imageWidth = this.data.metadata.originalWidth;
+    if (this.imageResizeRatio === 1 &&
+      imageWidth > this.OUTPUT_IMAGE_MAX_WIDTH_PX) {
+      return 'This image has been automatically downsized to ensure ' +
+        'that it will fit in the card.';
```

Comment:
```suggestion
      return (
        'This image has been automatically downsized to ensure '
        'that it will fit in the card.'
      );
```

Reply:
Done.